### PR TITLE
Detect glTF 2.0 and change forward to match Cesium's convention.

### PR DESCRIPTION
**Pull request by emackey**
I'm opening this a bit early to get Travis running on the tests, they seem to have problems.  Also, feedback welcome, particularly on the approach here.

I've added a new `forwardAxis` setting, similar to the existing private `upAxis` setting, on Model.js.  A change to the underlying gltf-pipeline marks models as having been upgraded from 1.0 (which includes models upgraded from 0.8 as they pass through 1.0 as an intermediate step).  Models which are *not* marked as upgraded are presumed to be native glTF 2.0 or higher, and thus must use the new +Z forward convention.  This is accomplished by adding an additional rotation, after the upAxis rotation for such models, to set the forward vector to align with what Cesium uses internally.

TODO:
- [x] Fix broken unit tests
- [x] Add unit test for new forwardAxis getter
- [x] CHANGES.md, breaking change (or rather breaking fix) for glTF 2.0 models only
- [x] Open PR into gltf-pipeline for the small change there: AnalyticalGraphicsInc/gltf-pipeline#371
- [x] Update 3D Tiles to set the old forward axis

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Signed CLA is on file.

@emackey, thanks for the pull request! Maintainers, we have a signed CLA from @emackey, so you can review this at any time.

:warning: I noticed that [CHANGES.md](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/CHANGES.md) has not been updated. If this change updates the public API in any way, fixes a bug, or makes any non-trivial update, please add a bullet point to `CHANGES.md` and comment on this pull request so we know it was updated. For more info, see the [Pull Request Guidelines]( https://github.com/AnalyticalGraphicsInc/cesium/blob/master/CONTRIBUTING.md#pull-request-guidelines).

:warning: I noticed that a file in one of our ThirdParty folders (`ThirdParty/, Source/ThirdParty/`) has been added or modified. Please verify that it has a section in [LICENSE.md](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/LICENSE.md) and that its license information is up to date with this new version. Once you do, please confirm by commenting on this pull request.

---

__I am a bot who helps you make Cesium awesome! [Contributions to my configuration are welcome.](https://github.com/AnalyticalGraphicsInc/cesium/tree/master/.concierge)__

:earth_africa: :earth_americas: :earth_asia:

**Comment by emackey** [comment]
Fixes #6591.

File: Apps/Sandcastle/gallery/Physically-Based Materials.html [hunk 1]
```diff
@@ -53,7 +53,7 @@
 viewer.scene.globe.depthTestAgainstTerrain = true;
 
 var position = new Cesium.Cartesian3(-1371108.6511167218, -5508684.080096612, 2901825.449865087);
-var heading = Cesium.Math.toRadians(90);
+var heading = Cesium.Math.toRadians(180);
```
Comment (emackey) [inline Apps/Sandcastle/gallery/Physically-Based Materials.html]: Point of trivia, this change actually puts the heading back to [what is was prior to #6592](https://github.com/AnalyticalGraphicsInc/cesium/pull/6592/files#diff-4ff54e0300165793e287a71858f79a7cL54), which is not a coincidence.  The model was updated to the correct glTF 2.0 orientation in that PR, but Cesium wasn't yet handling the 2.0 orientations correctly, so this was a workaround that is now being removed.

**Comment by lilleyse** [comment]
For glTF 2.0 in general this is probably the right thing to do.

I'm a little worried about how this affects 3D Tiles, where unlike a standalone model there isn't really a concept of a forward axis. Maybe b3dm/i3dm should always pass in `Axis.X` just so the content isn't rotated.

File: Source/Scene/Model.js [hunk 2]
```diff
@@ -579,6 +579,7 @@ define([
         this._ignoreCommands = defaultValue(options.ignoreCommands, false);
         this._requestType = options.requestType;
         this._upAxis = defaultValue(options.upAxis, Axis.Y);
+        this._forwardAxis = defaultValue(options.upAxis, Axis.Z);
```
Comment (lilleyse) [inline Source/Scene/Model.js]: Should this be:

`this._forwardAxis = defaultValue(options.forwardAxis, Axis.Z);`

Comment (emackey) [reply Source/Scene/Model.js]: Thanks!

**Comment by emackey** [comment]
We definitely need to address 3D Tiles before this gets merged.  It looks completely broken.

**Comment by lilleyse** [comment]
@emackey I'll aim to look at this closer after the release today.

**Comment by lilleyse** [comment]
Thanks @emackey, the Sandcastle demos look good. I substituted the ground vehicle for the milk truck in Multi-part CZML, which had the wrong orientation in master but is fixed here.

I moved the breaking change to the 1.47 section to CHANGES.md.

**Comment by lilleyse** [comment]
@emackey what do you think about having i3dm tiles also pass in their forward axis as `X`, like b3dm. The issue of auto-rotated instanced models came up on the forum - https://groups.google.com/forum/#!topic/cesium-dev/w5DnTHZATRo

**Comment by emackey** [comment]
Yes, certainly.  Sadly I'm not a 3D Tiles expert (yet) so I'm sure I just overlooked some of the constructors.  Are there any others?

**Comment by lilleyse** [comment]
That should be it. Thanks!

**Pull request merged**